_posts/2015-07-08-elasticsearch-is-coming.md

```diff
@@ -0,0 +1,798 @@
+modified: 2015-07-15
+tags: [gitlab, git]
+comments: true
+image:
+  feature: bandeau-gitlab.jpg
+  credit: Alban Pommeret
+  creditlink: http://reputationvip.io
+---
+
+# ELASTICSEARCH IS COMING
+
+## ROADMAP
+
+This article is the first of many to come about Elasticsearch. Elasticsearch is a vast and amazing technology, and I could go on talking about it for hours.
+
+So first, let me introduce the summary of this very first article:
+
+1. A smooth introduction to Elasticsearch
+2. Theoretical approach: What's under the armor of Elasticsearch ?
+3. Installation and configuration of Elasticsearch
+4. Querying: Basic CRUD
+5. Introduction to plugins
+
+***
+
+## How to read these articles ?
+
+Before we dive in Elasticsearch head first, let me tell you how I organized these articles.
+
+First of all, this first article is about the very basics of Elasticsearch, and I will not really talk about full-text research right now. The main reason is that I consider
+that you need to take a look at the basic operations first.
+
+**But don't be afraid, I will talk about the full-text research in the next article**.
+
+#### The articles
+
+First of all, throughout these articles, I will guide you through the understanding of Elasticsearch. 
+Each part of the articles will provide you theoretical explanations, as well as examples.
+
+At ReputationVIP, we are huge fans of Game of Thrones. So, the guideline I chose for this set of articles is Game of Thrones.
+All along these articles (though I don't know how many of them will follow this one), we will build a database filled with the Game of Thrones characters.
+
+#### The Github Repository
+
+Available here: [https://github.com/quentinfayet/elasticsearch](https://github.com/quentinfayet/elasticsearch)
+
+It contains everything you need to run the articles' examples: A ready-to-run Elasticsearch cluster under Docker containers, available to run with Docker-Compose.
+
+#### The Docker containers
+
+Beside being a big fan of Game of Thrones, I also am a big fan of Docker. For those who don't know about Docker, you can take a look here: [https://www.docker.com/](https://www.docker.com/).
+So, I decided to create an Elasticsearch **cluster** with Docker containers.
+Each **node** of this **cluster** takes the name of a member of the Stark family.
+
+In deed, I know how difficult it can be to read and understand other people's examples, but with this Docker container, you will be in the same environment as I used to write this article.
+
+To launch the docker cluster, first clone the repository in the folder of your choice, and go into it.
+
+I also require you to install [https://docs.docker.com/compose/]( *docker compose*), which is a good solution to build and run multi-container applications with Docker.
+The installation less than takes 2 minutes, and is really simple to use. Go here to install it: [https://docs.docker.com/compose/install/](https://docs.docker.com/compose/install/)
```

Comment:
I didn't know about that particular way to write it, never heard about it. Anyway, I will change it to `[...] takes less than [...]`